codegen/src/main/kotlin/apifi/codegen/OperationExtension.kt

```diff
@@ -40,6 +36,12 @@ fun Operation.requestParamNames(modelMapping: Map<String, String>) = request?.le
             .build())
 }?.map { it.name } ?: emptyList()
 
-fun Operation.returnType(modelMapping: Map<String, String>) = responses
-        ?.firstOrNull { it.defaultOrStatus == "200" || it.defaultOrStatus == "201" }
-        ?.let { ClassName(micronautHttpPackage, "HttpResponse").parameterizedBy(it.type.toKotlinPoetType(modelMapping)) }
\ No newline at end of file
+fun Operation.returnType(modelMapping: Map<String, String>): ParameterizedTypeName? =
+    responses.let { res ->
+        when {
+            hasOnlyDefaultResponse() -> ClassName(micronautHttpPackage, "HttpResponse").parameterizedBy(res[0].type.toKotlinPoetType(modelMapping))
+            hasMoreThanOne2xxResponse() -> error("Invalid responses defined for operation with identifier: ${this.name}. Has more than one 2xx responses defined")
+            else -> first2xxResponse()?.let { ClassName(micronautHttpPackage, "HttpResponse").parameterizedBy(it.type.toKotlinPoetType(modelMapping)) }
+        }
+    }
+//TODO: we could give functionality of marking an operation as preferred one, in case we want to support multiple 2xx for some reason
```

Comment:
@kdabir Right now if an operation has more than one 2xx status defined, I am throwing an exception. Added a TODO to see if there would be cases where this might be needed and in that case we could have a label to identify which is preferred or default one.